docs/DeveloperGuide.md

```diff
@@ -1,29 +1,204 @@
 # Developer Guide
 
 ## Acknowledgements
+Credits to [Personbook](https://github.com/nus-cs2113-AY2223S2/personbook)
+for some reused skeleton code and inspiration on OOP implementation.
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Setting up
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Design 
+### Architecture
+![Architecture_Diagram](uml/images/DinerDirectorArchitectureDiagram.png)
+The **Architecture Diagram** given above explains the high-level design of DinerDirector. 
+Given below is a quick overview of main components and how they interact with each other. 
 
+**Architecture of Main Components** <br/>
+`DinerDirector` entry point to the application is DinerDirector. Initially, `DinerDirector` class interact with `UI` class to 
+prompt user for inputs. Once user keys in the input text, the input will be redirected to `Parser` class. `Parser` class will manage the error 
+handling of user's input. Next, any logic will be executed by `Command` object. If there exists any interaction with `Entity`, `Command` will 
+request to `Manager` to assist the CRUD operations.  
+
+### Command Component (Zheng Rong)
+
+##### The command component consists of the following:
+
+![Command_Package_Diagram](uml/images/CommandPackageDiagram.png)
+
+- 4 subcomponents: Command, HelpCommand, ExitCommand, IncorrectCommand
+
+- 4 sub packages with the following names: deadline, meeting, menu and staff
+    - Each package contains the 3 general commands:
+        - AddCommand
+        - DeleteCommand
+        - ViewCommand
+    - The general commands have specific names depending on the package it is in.
+
+The command component consist of the Abstract `Command` class.
+It has multiple child `Command` classes that inherits from the Abstract `Command` Class.
+Each child `Command` class accepts arguments that are parsed from the `Parser` class and set those arguments to private variables in their respective classes, if there is any private variables to be set.
+An `execute` method will execute the operations needed for each individual commands.
+An `isExit` method will return a boolean value that decides whether the program should exit after this command is called.
+
+
+### Manager Component (PeiHao)
```

Comment:
I would recommend removing your names from the headings! Makes it look more cohesive